python/rascaline/tests/utils/splines.py

```diff
@@ -102,3 +111,261 @@ def test_splines_numerical_derivative_error():
     match = "Numerically derivative of the radial integral can not be performed"
     with pytest.raises(ValueError, match=match):
         RadialIntegralFromFunction(**kwargs).compute()
+
+
+def test_kspace_radial_integral():
+    """Test against anayliycal integral with Gaussian densities and GTOs"""
+
+    cutoff = 2
+    max_radial = 6
+    max_angular = 3
+    atomic_gaussian_width = 1.0
+    k_cutoff = 1.2 * np.pi / atomic_gaussian_width
+
+    basis = GtoBasis(cutoff=cutoff, max_radial=max_radial)
+
+    spliner = KSpaceSpliner(
+        max_radial=max_radial,
+        max_angular=max_angular,
+        k_cutoff=k_cutoff,
+        basis=basis,
+        density=DeltaDensity(),  # density does not enter in a Kspace radial integral
+        accuracy=1e-8,
+    )
+
+    Neval = 100
+    kk = np.linspace(0, k_cutoff, Neval)
+
+    sigma = np.ones(max_radial, dtype=float)
+    for i in range(1, max_radial):
+        sigma[i] = np.sqrt(i)
+    sigma *= cutoff / max_radial
+
+    factors = np.sqrt(np.pi) * np.ones((max_radial, max_angular + 1))
+
+    coeffs_num = np.zeros([max_radial, max_angular + 1, Neval])
+    coeffs_exact = np.zeros_like(coeffs_num)
+
+    for ell in range(max_angular + 1):
+        for n in range(max_radial):
+            i1 = 0.5 * (3 + n + ell)
+            i2 = 1.5 + ell
+            factors[n, ell] *= (
+                2 ** (0.5 * (n - ell - 1))
+                * gamma(i1)
+                / gamma(i2)
+                * sigma[n] ** (2 * i1)
+            )
+            coeffs_exact[n, ell] = (
+                factors[n, ell]
+                * kk**ell
+                * hyp1f1(i1, i2, -0.5 * (kk * sigma[n]) ** 2)
+            )
+
+            coeffs_num[n, ell] = spliner._radial_integral(n, ell, kk)
+
+    assert_allclose(coeffs_num, coeffs_exact)
+
+
+# def test_rspace_radial_integral():
```

Comment:
why is this commented out?

Reply:
Becuase, I still need to fix the analytical expression. 🫣